lib/p2p/Parser.ts

```diff
@@ -60,20 +60,30 @@ interface Parser {
 
 /** Protocol packet parser */
 class Parser extends EventEmitter {
+  private pending = '';
+
+  constructor(private delimiter: string) {
+    super();
+  }
+
   public feed = (data: Buffer): void => {
-    const dataStr = data.toString();
-    const packetLines = dataStr.split('\r\n');
-    packetLines.forEach((packetStr) => {
-      if (!packetStr) {
-        return;
-      }
-      try {
-        const packet = fromRaw(packetStr);
-        this.emit('packet', packet);
-      } catch (err) {
-        this.emit('error', err);
-      }
-    });
+    this.pending += data.toString();
+    const index = this.pending.indexOf(this.delimiter);
+    if (index > -1) {
```

Comment:
Suggest to have tests for:
- empty Buffer
- buffer containing one delimiter only (no other data)
- buffer containing 5  delimiters (no other data)